src/Microsoft.DotNet.Interactive.HttpRequest.Tests/ParserTests.cs

```diff
@@ -0,0 +1,614 @@
+                <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+            </request>
+            """);
+
+        using var _ = new AssertionScope();
+
+        var requestNode = result.SyntaxTree.RootNode
+              .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+        requestNode.BodyNode.Text.Should().Be(
+            """
+            <request>
+                <name>sample</name>
+                <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+            </request>
+            """);
+    }
+
+    [Fact]
+    public void multiple_new_lines_before_body_are_parsed_correctly()
+    {
+        var result = HttpRequestParser.Parse(
+            """
+            POST https://example.com/comments HTTP/1.1
+            Content-Type: application/xml
+            Authorization: token xxx
+
+
+
+
+            <request>
+                <name>sample</name>
+                <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+            </request>
+            """);
+
+        using var _ = new AssertionScope();
+
+        var requestNode = result.SyntaxTree.RootNode
+              .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+        
+        requestNode.BodyNode.Text.Should().Be(
+            """
+            <request>
+                <name>sample</name>
+                <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+            </request>
+            """);
+    }
+
+    [Fact]
+    public void comments_are_parsed_correctly()
+    {
+
+        var result = HttpRequestParser.Parse(
+            """
+            # This is a comment
+            GET https://example.com HTTP/1.1"
+            """);
+
+        using var _ = new AssertionScope();
+
+        var methodNode = result.SyntaxTree.RootNode
+              .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+              .MethodNode;
+
+        methodNode.ChildNodes.Should().ContainSingle<HttpCommentNode>().Which.Text.Should().Be(
+          "# This is a comment");
+
+              
+    }
+
+    [Fact]
+    public void diagnostic_object_is_reported_for_unrecognized_verb()
+    {
+        var result = HttpRequestParser.Parse("OOOOPS https://example.com");
+
+        result.GetDiagnostics()
+            .Should().ContainSingle().Which.Message.Should().Be("Unrecognized HTTP verb OOOOPS");
+    }
+
+    [Fact]
+    public void request_node_without_method_node_created_correctly()
+    {
+        var result = HttpRequestParser.Parse("https://example.com");
+
+        result.SyntaxTree.RootNode.ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+            .MethodNode.Should().BeNull();
+    }
+
+    [Fact]
+    public void request_node_without_method_node_created()
+    {
+        var result = HttpRequestParser.Parse("https://example.com");
+
+        result.SyntaxTree.RootNode.ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+            .MethodNode.Should().BeNull();
+    }
+}
+// TODO: Test string with variable declarations but no requests
```

Comment:
Do we need the commented out bit below for anything?

Reply:
The commented bit is the set of tests copied over from the old parser - we were cutting and pasting individual tests from below to above.

I think we may want to preserve this until we have tests that test the new parser's handling for the commented cases...

Reply:
Understood. We should at least delete the ones that represent unfinished work.

Reply:
Yes, I think there may be some scenarios pertaining to features we have already implemented that would be nice to move above now as opposed to later.